pkg/server/httpserver.go

```diff
@@ -111,6 +111,19 @@ func FormatAndReturnTextResponse(res http.ResponseWriter, data string) {
 	return
 }
 
+// FormatJSONAndReturnTextResponse formats the given data into JSON and returns a plaintext response
+func FormatJSONAndReturnTextResponse(res http.ResponseWriter, data interface{}) {
```

Comment:
[nit] the name makes it sound like you're taking JSON as an input and converting it to text. I think `FormatAndReturnJSONTextResponse` would be a better name

Reply:
on it